---
layout: tutorial_hands_on

title: Clinical-MP-Database generation
zenodo_link: 'https://doi.org/10.5281/zenodo.8144258'
level: Intermediate
questions:
- How do I generate a large database for metaproteomics analysis?
- How can I create a compact database from the comprehensive database so that I don't face issues while performing database searching?
objectives:
- Ways to download protein database
- Generation of compact database
time_estimation: 3H
key_points:
- download the protein database with taxon names
contributors:
- subinamehta
- dechendb
- PratikDJagtap
- katherine-d21
  
follow_up_training:

    -
        type: "internal"
        topic_name: proteomics
        tutorials:
            - clinical-metaproteomics-discovery
            - clinical-metaproteomics-validation
            - clinical-metaproteomics-quantitation
            - clinical-metaproteomics-data-interpretation

subtopic: multi-omics
tags: [microbiome]
---
---

# Introduction

<!-- This is a comment. -->

Metaproteomics {% cite Metaproteomics_video %}  is the large-scale characterization of the entire complement of proteins expressed by microbiota. However, metaproteomics analysis of clinical samples {% cite Metaproteomics_community_effort %} {% cite Jagtap2015 %} is challenged by the presence of abundant human (host) proteins which hampers the confident detection of lower abundant microbial proteins. To address this, we used tandem mass spectrometry (MS/MS) and bioinformatics tools on the Galaxy platform to develop a metaproteomics workflow to characterize the metaproteomes of clinical samples. This clinical metaproteomics workflow holds potential for general clinical applications such as potential secondary infections during COVID infection, microbiome changes during cystic fibrosis as well as broad research questions regarding host-microbe interactions. The Galaxy-P team has developed a workflow wherein a large database is generated by downloading protein sequences of known disease-causing micro-organisms (figure 1)and  then generating a compact database from the comprehensive database using the Metanovo tool.

![Workflow-1](../../images/cmp-figure1.png){: width="75%"}

In this tutorial, as an example, we will analyze a subset of pap-smear fluid (PTF) samples collected from ovarian cancer (OC) and non-OC patients. For this pilot study, proteins were isolated from each sample, digested into peptides, and labeled with a distinct Tandem Mass Tag (TMT)-11-plex tagging reagent. The samples were then separated by offline high pH reversed-phase liquid chromatography (RPLC) and were analyzed using a hybrid quadrupole-Orbitrap mass spectrometer.

> <agenda-title></agenda-title>
>
> In this tutorial, we will cover:
>
> 1. TOC
> {:toc}
>
{: .agenda}


# Pretreatment
## Get data

> <hands-on-title> Data Upload </hands-on-title>
>
> 1. Create a new history for this tutorial
>
>    {% snippet faqs/galaxy/histories_create_new.md %}
>
>    {% snippet faqs/galaxy/histories_rename.md %}
>
> 2. Import the files: 5 MGF files, 3 tabular files from [Zenodo]({{ page.zenodo_link }}) or from
>    the shared data library (`GTN - Material` -> `{{ page.topic_name }}`
>     -> `{{ page.title }}`):
>
>    ```
>    https://zenodo.org/record/8144258/files/Genera_tabular.tabular
>    https://zenodo.org/record/8144258/files/Order_Family_tabular.tabular
>    https://zenodo.org/record/8144258/files/Species_tabular.tabular
>    https://zenodo.org/record/8144258/files/PTRC_Skubitz_Plex2_F10_9Aug19_Rage_Rep-19-06-08.mgf
>    https://zenodo.org/record/8144258/files/PTRC_Skubitz_Plex2_F11_9Aug19_Rage_Rep-19-06-08.mgf
>    https://zenodo.org/record/8144258/files/PTRC_Skubitz_Plex2_F13_9Aug19_Rage_Rep-19-06-08.mgf
>    https://zenodo.org/record/8144258/files/PTRC_Skubitz_Plex2_F14_9Aug19_Rage_Rep-19-06-08.mgf
>    https://zenodo.org/record/8144258/files/PTRC_Skubitz_Plex2_F15_9Aug19_Rage_Rep-19-06-08.mgf
>
>    ```
>
>    {% snippet faqs/galaxy/datasets_import_via_link.md %}
>
>    {% snippet faqs/galaxy/datasets_import_from_data_library.md %}
>
>
>
> 3. Rename the datasets
> 4. Check that the datatype
>
>    {% snippet faqs/galaxy/datasets_change_datatype.md datatype="datatypes" %}
>
> 5. Optional-Add to each database a tag corresponding to the file name.
>    {% snippet faqs/galaxy/datasets_add_tag.md %}
>
> 6. Build a **Dataset list** for the 5 MGF files. Rename the dataset collection as `MGF dataset collection`.
>
{: .hands_on}

# We have a choice to run all these steps using a single workflow, then discuss each step and the results in more detail.

> <hands-on-title>Running the Workflow</hands-on-title>
>
> 1. **Import the workflow** into Galaxy:
>
>    {% snippet faqs/galaxy/workflows_run_trs.md path="topics/proteomics/tutorials/clinical-mp-database-generation/workflows/main_workflow.ga" title="Pretreatments" %}
>
>
>
> 2. Run **Workflow** {% icon workflow %} using the following parameters:
>    - *"Send results to a new history"*: `No`
>    - {% icon param-file %} *"1: Input Dataset collection"*: `MGF dataset collection`
>    - {% icon param-file %} *"2: General_tabular"*: `Genera_tabular.tabular`
>    - {% icon param-file %} *"3: Species_tabular"*: `Species_tabular.tabular`
>    - {% icon param-file %} *"4: Order_Family_tabular "*:`Order_Family_tabular.tabular`
>
>    {% snippet faqs/galaxy/workflows_run.md %}
>
{: .hands_on}

# Step-by-step analysis

# Download Protein Sequences using taxon names

## Sub-step with **UniProt**

> <hands-on-title> Download Protein Sequences using UniProt XML downloader</hands-on-title>
>
> 1. {% tool [UniProt](toolshed.g2.bx.psu.edu/repos/galaxyp/uniprotxml_downloader/uniprotxml_downloader/2.3.0) %} with the following parameters:
>    - *"Select"*: `Genera_tabular.tabular`
>        - {% icon param-file %} *"Dataset (tab separated) with Taxon ID/Name column"*: `output` (Input dataset)
>        - *"Column with Taxon ID/name"*: `c1`
>    - *"uniprot output format"*: `fasta`
> 2. Rename the output as Genera.fasta
>
>    > <comment-title> short description </comment-title>
>    >
>    > This tool will help download the protein fasta sequences by inputting the taxon names.
>    {: .comment}
>
{: .hands_on}

## Sub-step with **UniProt**

> <hands-on-title> Task description </hands-on-title>
>
> 1. {% tool [UniProt](toolshed.g2.bx.psu.edu/repos/galaxyp/uniprotxml_downloader/uniprotxml_downloader/2.3.0) %} with the following parameters:
>    - *"Select"*: `Species_tabular.tabular`
>        - {% icon param-file %} *"Dataset (tab separated) with Taxon ID/Name column"*: `output` (Input dataset)
>        - *"Column with Taxon ID/name"*: `c1`
>    - *"uniprot output format"*: `fasta`
> 2. Rename the output as Species.fasta
>

## Sub-step with **UniProt**

> <hands-on-title> Task description </hands-on-title>
>
> 1. {% tool [UniProt](toolshed.g2.bx.psu.edu/repos/galaxyp/uniprotxml_downloader/uniprotxml_downloader/2.3.0) %} with the following parameters:
>    - *"Select"*: `Order_Family_tabular.tabular`
>        - {% icon param-file %} *"Dataset (tab separated) with Taxon ID/Name column"*: `output` (Input dataset)
>        - *"Column with Taxon ID/name"*: `c1`
>    - *"uniprot output format"*: `fasta`
> 2. Rename the output as Order_Family.fasta
>
{: .hands_on}

> <question-title></question-title>
>
> 1. Why are we using the tools separately? Can we run it all together?
>
> > <solution-title></solution-title>
> >
> > 1. The tools are run separately to reduce the load on the server and tool. If you have a limited number of taxon names then you can run it all together.
> >
> {: .solution}
>
{: .question}

## Sub-step with **FASTA Merge Files and Filter Unique Sequences**

> <hands-on-title> Task description </hands-on-title>
>
> 1. {% tool [FASTA Merge Files and Filter Unique Sequences](toolshed.g2.bx.psu.edu/repos/galaxyp/fasta_merge_files_and_filter_unique_sequences/fasta_merge_files_and_filter_unique_sequences/1.2.0) %} with the following parameters:
>    - *"Run in batch mode?"*: `Merge individual FASTAs (output collection if the input is collection)`
>        - In *"Input FASTA File(s)"*:
>            - {% icon param-repeat %} *"Insert Input FASTA File(s)"*
>                - {% icon param-file %} *"FASTA File"*: `Genera.fasta` (output of **UniProt** {% icon tool %})
>            - {% icon param-repeat %} *"Insert Input FASTA File(s)"*
>                - {% icon param-file %} *"FASTA File"*: `Species.fasta` (output of **UniProt** {% icon tool %})
>            - {% icon param-repeat %} *"Insert Input FASTA File(s)"*
>                - {% icon param-file %} *"FASTA File"*: `Order_family.fasta` (output of **UniProt** {% icon tool %})
>
>
>    > <comment-title> short description </comment-title>
>    >
>    > This tool merges all the 3 FASTA databases from Uniprot XML download tool.
>    {: .comment}
>
{: .hands_on}
> <question-title></question-title>
>
> 1. Can we select multiple files together?
> 2. Question2?
>
> > <solution-title></solution-title>
> >
> > 1. Yes, that certainly can be done. We used one input file at a time to maintain the order of sequences in the database.
> > 2. Answer for question2
> >
> {: .solution}
>
{: .question}

## Sub-step with **Protein Database Downloader**

> <hands-on-title> Task description </hands-on-title>

>
> 1. {% tool [Protein Database Downloader](toolshed.g2.bx.psu.edu/repos/galaxyp/dbbuilder/dbbuilder/0.3.4) %} with the following parameters:
>    - *"Dowload from?"*: `UniProtKB`
>        - In *"Taxonomy"*: `Homo sapiens (Human)`
>	 - In  *"Proteome Set"*: `Reference Proteome Set`
>	 - In  *"Include isoform data"*: `Yes`
>2. Rename as "Protein Database Human UniProt with Isoforms"
{: .hands_on}
## Sub-step with **Protein Database Downloader**

> <hands-on-title> Task description </hands-on-title>

>
> 1. {% tool [Protein Database Downloader](toolshed.g2.bx.psu.edu/repos/galaxyp/dbbuilder/dbbuilder/0.3.4) %} with the following parameters:
>    - *"Dowload from?"*: `cRAP (contaminants)`
>2. Rename as "Protein Database Contaminants (cRAP)"
>`
>
{: .hands_on}
## Sub-step with **FASTA Merge Files and Filter Unique Sequences**

> <hands-on-title> Task description </hands-on-title>
>
> 1. {% tool [FASTA Merge Files and Filter Unique Sequences](toolshed.g2.bx.psu.edu/repos/galaxyp/fasta_merge_files_and_filter_unique_sequences/fasta_merge_files_and_filter_unique_sequences/1.2.0) %} with the following parameters:
>    - *"Run in batch mode?"*: `Merge individual FASTAs (output collection if input is collection)`
>        - In *"Input FASTA File(s)"*:
>            - {% icon param-repeat %} *"Insert Input FASTA File(s)"*
>                - {% icon param-file %} *"FASTA File"*: `Protein Database Human UniProt with Isoforms` (output of **Protein Database Downloader** {% icon tool %})
>                - {% icon param-file %} *"FASTA File"*: `Protein Database Contaminants (cRAP)` (output of **Protein Database Downloader** {% icon tool %})
{: .hands_on}
# Reducing Database size

## Sub-step with **MetaNovo**

> <hands-on-title> Metanovo tool generates a compact database from your comprehensive database. </hands-on-title>
>
> 1. {% tool [MetaNovo](toolshed.g2.bx.psu.edu/repos/galaxyp/metanovo/metanovo/1.9.4+galaxy4) %} with the following parameters:
>    - *"MGF Input Type"*: `Collection`
>        - {% icon param-collection %} *"MGF Collection"*: `output` (Input dataset collection)
>    - {% icon param-file %} *"FASTA File"*: `output` (output of **FASTA Merge Files and Filter Unique Sequences** {% icon tool %})
>    - In *"Spectrum Matching Parameters"*:
>        - *"Precursor ion mass tolerance"*: `10.0`
>        - *"Precursor ion tolerance unit"*: `ppm`
>        - *"Fragment ion mass tolerance"*: `0.01`
>        - *"Fragment ion tolerance unit"*: `Da`
>        - *"Enzyme"*: `Trypsin (no P Rule)`
>        - *"Fixed modifications as comma separated list"*: `Carbamidomethylation of C, TMT 10-plex of K, TMT 10-plex of peptide N term`
>        - *"Variable modifications as comma separated list"*: `Oxidation of M`
>        - *"Maximal charge to search for"*: `5`
>    - In *"Import Filters"*:
>        - *"The maximal peptide length to consider when importing identification files"*: `50`
> 2. Rename the output as "Metanovo compact database"
{: .hands_on}


> <question-title></question-title>
>
> 1. Why are we reducing the size of the database?
> 2. Why is this running TMT10 plex modification when the data is 11-plex?
>
> > <solution-title></solution-title>
> >
> > 1. Reducing the size of the database improves search speed, FDR and sensitivity.
> > 2. There is no option for 11-plex modifications in Metanovo, hence we use the TMT-10plex.
> >
> {: .solution}
>
{: .question}

## Sub-step with **Protein Database Downloader**

> <hands-on-title> Task description </hands-on-title>
> 1. {% tool [Protein Database Downloader](toolshed.g2.bx.psu.edu/repos/galaxyp/dbbuilder/dbbuilder/0.3.4) %} with the following parameters:
>    - *"Dowload from?"*: `UniProtKB(reviewed only)`
>        - In *"Taxonomy"*: `Homo sapiens (Human)`
>	 - In  *"Proteome Set"*: `Reference Proteome Set`
>	 - In  *"Include isoform data"*: `Yes`
>2. Rename as "Protein Database Human SwissProt".
{: .hands_on}

## Sub-step with **FASTA Merge Files and Filter Unique Sequences**

> <hands-on-title> Task description </hands-on-title>
>
> 1. {% tool [FASTA Merge Files and Filter Unique Sequences](toolshed.g2.bx.psu.edu/repos/galaxyp/fasta_merge_files_and_filter_unique_sequences/fasta_merge_files_and_filter_unique_sequences/1.2.0) %} with the following parameters:
>    - *"Run in batch mode?"*: `Merge individual FASTAs (output collection if input is collection)`
>        - In *"Input FASTA File(s)"*:
>            - {% icon param-repeat %} *"Insert Input FASTA File(s)"*
>                - {% icon param-file %} *"FASTA File"*: `Protein Database Human SwissProt"` (output of **Protein Database downloader** {% icon tool %})
>                - {% icon param-file %} *"FASTA File"*: `cRAP (contaminants)` (output of **Protein Database downloader**{% icon tool %})
>                - {% icon param-file %} *"FASTA File"*: `Metanovo compact database` (output of **MetaNovo** {% icon tool %})
>
>
>    > <comment-title> short description </comment-title>
>    >
>    > A comment about the tool or something else. This box can also be in the main text
>    {: .comment}
>
{: .hands_on}


# Conclusion

The first step for the Clinical Metaproteomics study is database generation. As we didnâ€™t have a reference database or information from 16srRNA-seq data, we generated a fasta database doing a literature survey, however, if 16D rRNA data is present, the taxon identified can be used for a customized database generation. As the size of the comprehensive database is generally  too large, we used the Metanovo tool to reduce the size of the database. This reduced database will be then used for clinical metaproteomics discovery workflow.
