---
layout: tutorial_hands_on

title: Clinical-MP-Database Generation
zenodo_link: ''
questions:
- Which biological questions are addressed by the tutorial?
- Which bioinformatics techniques are important to know for this type of data?
objectives:
- The learning objectives are the goals of the tutorial
- They will be informed by your audience and will communicate to them and to yourself
  what you should focus on during the course
- They are single sentences describing what a learner should be able to do once they
  have completed the tutorial
- You can use Bloom's Taxonomy to write effective learning objectives
time_estimation: 3H
key_points:
<<<<<<< Updated upstream
- download the protein database with taxon names
contributors:
- subinamehta
- dechendb
- PratikDJagtap
- katherine-d21
  
follow_up_training:

    -
        type: "internal"
        topic_name: proteomics
        tutorials:
            - clinical-metaproteomics-discovery
            - clinical-metaproteomics-validation
            - clinical-metaproteomics-quantitation
            - clinical-metaproteomics-data-interpretation
=======
- The take-home messages
- They will appear at the end of the tutorial
contributors:
- contributor1
- contributor2

---
>>>>>>> Stashed changes


# Introduction

<!-- This is a comment. -->

<<<<<<< Updated upstream
Metaproteomics {% cite Metaproteomics_video %}  is the large-scale characterization of the entire complement of proteins expressed by microbiota. However, metaproteomics analysis of clinical samples {% cite Metaproteomics_community_effort %} {% cite Jagtap2015 %} is challenged by the presence of abundant human (host) proteins which hampers the confident detection of lower abundant microbial proteins. To address this, we used tandem mass spectrometry (MS/MS) and bioinformatics tools on the Galaxy platform to develop a metaproteomics workflow to characterize the metaproteomes of clinical samples. This clinical metaproteomics workflow holds potential for general clinical applications such as potential secondary infections during COVID infection, microbiome changes during cystic fibrosis as well as broad research questions regarding host-microbe interactions. The Galaxy-P team has developed a workflow wherein a large database is generated by downloading protein sequences of known disease-causing micro-organisms (figure 1)and  then generating a compact database from the comprehensive database using the Metanovo tool.
=======
General introduction about the topic and then an introduction of the
tutorial (the questions and the objectives). It is nice also to have a
scheme to sum up the pipeline used during the tutorial. The idea is to
give to trainees insight into the content of the tutorial and the (theoretical
and technical) key concepts they will learn.
>>>>>>> Stashed changes

You may want to cite some publications; this can be done by adding citations to the
bibliography file (`tutorial.bib` file next to your `tutorial.md` file). These citations
must be in bibtex format. If you have the DOI for the paper you wish to cite, you can
get the corresponding bibtex entry using [doi2bib.org](https://doi2bib.org).

With the example you will find in the `tutorial.bib` file, you can add a citation to
this article here in your tutorial like this:
{% raw %} `{% cite Batut2018 %}`{% endraw %}.
This will be rendered like this: {% cite Batut2018 %}, and links to a
[bibliography section](#bibliography) which will automatically be created at the end of the
tutorial.

<<<<<<< Updated upstream
=======

**Please follow our
[tutorial to learn how to fill the Markdown]({{ site.baseurl }}/topics/contributing/tutorials/create-new-tutorial-content/tutorial.html)**

>>>>>>> Stashed changes
> <agenda-title></agenda-title>
>
> In this tutorial, we will cover:
>
> 1. TOC
> {:toc}
>
{: .agenda}

# Title for your first section

<<<<<<< Updated upstream
# Pretreatment
=======
Give some background about what the trainees will be doing in the section.
Remember that many people reading your materials will likely be novices,
so make sure to explain all the relevant concepts.

## Title for a subsection
Section and subsection titles will be displayed in the tutorial index on the left side of
the page, so try to make them informative and concise!

# Hands-on Sections
Below are a series of hand-on boxes, one for each tool in your workflow file.
Often you may wish to combine several boxes into one or make other adjustments such
as breaking the tutorial into sections, we encourage you to make such changes as you
see fit, this is just a starting point :)

Anywhere you find the word "***TODO***", there is something that needs to be changed
depending on the specifics of your tutorial.

have fun!

>>>>>>> Stashed changes
## Get data

> <hands-on-title> Data Upload </hands-on-title>
>
> 1. Create a new history for this tutorial
> 2. Import the files from [Zenodo]({{ page.zenodo_link }}) or from
>    the shared data library (`GTN - Material` -> `{{ page.topic_name }}`
>     -> `{{ page.title }}`):
>
>    ```
<<<<<<< Updated upstream
>    https://zenodo.org/record/8144258/files/Genera_tabular.tabular
>    https://zenodo.org/record/8144258/files/Order_Family_tabular.tabular
>    https://zenodo.org/record/8144258/files/Species_tabular.tabular
>    https://zenodo.org/record/8144258/files/PTRC_Skubitz_Plex2_F10_9Aug19_Rage_Rep-19-06-08.mgf
>    https://zenodo.org/record/8144258/files/PTRC_Skubitz_Plex2_F11_9Aug19_Rage_Rep-19-06-08.mgf
>    https://zenodo.org/record/8144258/files/PTRC_Skubitz_Plex2_F13_9Aug19_Rage_Rep-19-06-08.mgf
>    https://zenodo.org/record/8144258/files/PTRC_Skubitz_Plex2_F14_9Aug19_Rage_Rep-19-06-08.mgf
>    https://zenodo.org/record/8144258/files/PTRC_Skubitz_Plex2_F15_9Aug19_Rage_Rep-19-06-08.mgf
>
=======
>    
>>>>>>> Stashed changes
>    ```
>    ***TODO***: *Add the files by the ones on Zenodo here (if not added)*
>
>    ***TODO***: *Remove the useless files (if added)*
>
>    {% snippet faqs/galaxy/datasets_import_via_link.md %}
>
>    {% snippet faqs/galaxy/datasets_import_from_data_library.md %}
>
> 3. Rename the datasets
> 4. Check that the datatype
>
>    {% snippet faqs/galaxy/datasets_change_datatype.md datatype="datatypes" %}
>
<<<<<<< Updated upstream
> 5. Optional-Add to each database a tag corresponding to the file name.
=======
> 5. Add to each database a tag corresponding to ...
>
>>>>>>> Stashed changes
>    {% snippet faqs/galaxy/datasets_add_tag.md %}
>
{: .hands_on}

# Title of the section usually corresponding to a big step in the analysis

<<<<<<< Updated upstream
> <hands-on-title>Running the Workflow</hands-on-title>
>
> 1. **Import the workflow** into Galaxy:
>
>    {% snippet faqs/galaxy/workflows_run_trs.md path="topics/proteomics/tutorials/clinical-mp-database-generation/workflows/main_workflow.ga" title="Pretreatments" %}
>
>
>
> 2. Run **Workflow** {% icon workflow %} using the following parameters:
>    - *"Send results to a new history"*: `No`
>    - {% icon param-file %} *"1: Input Dataset collection"*: `MGF dataset collection`
>    - {% icon param-file %} *"2: General_tabular"*: `Genera_tabular.tabular`
>    - {% icon param-file %} *"3: Species_tabular"*: `Species_tabular.tabular`
>    - {% icon param-file %} *"4: Order_Family_tabular "*:`Order_Family_tabular.tabular`
>
>    {% snippet faqs/galaxy/workflows_run.md %}
>
{: .hands_on}

# Step-by-step analysis

# Download Protein Sequences using taxon names

## Sub-step with **UniProt**

> <hands-on-title> Download Protein Sequences using UniProt XML downloader</hands-on-title>
>
> 1. {% tool [UniProt](toolshed.g2.bx.psu.edu/repos/galaxyp/uniprotxml_downloader/uniprotxml_downloader/2.3.0) %} with the following parameters:
>    - *"Select"*: `Genera_tabular.tabular`
>        - {% icon param-file %} *"Dataset (tab separated) with Taxon ID/Name column"*: `output` (Input dataset)
>        - *"Column with Taxon ID/name"*: `c1`
>    - *"uniprot output format"*: `fasta`
> 2. Rename the output as Genera.fasta
>
>    > <comment-title> short description </comment-title>
>    >
>    > This tool will help download the protein fasta sequences by inputting the taxon names.
>    {: .comment}
>
{: .hands_on}
=======
It comes first a description of the step: some background and some theory.
Some image can be added there to support the theory explanation:

![Alternative text](../../images/image_name "Legend of the image")

The idea is to keep the theory description before quite simple to focus more on the practical part.
>>>>>>> Stashed changes

***TODO***: *Consider adding a detail box to expand the theory*

> <details-title> More details about the theory </details-title>
>
> But to describe more details, it is possible to use the detail boxes which are expandable
>
<<<<<<< Updated upstream
{: .hands_on}
## Sub-step with **UniProt**

> <hands-on-title> Task description </hands-on-title>
>
> 1. {% tool [UniProt](toolshed.g2.bx.psu.edu/repos/galaxyp/uniprotxml_downloader/uniprotxml_downloader/2.3.0) %} with the following parameters:
>    - *"Select"*: `Order_Family_tabular.tabular`
>        - {% icon param-file %} *"Dataset (tab separated) with Taxon ID/Name column"*: `output` (Input dataset)
>        - *"Column with Taxon ID/name"*: `c1`
>    - *"uniprot output format"*: `fasta`
> 2. Rename the output as Order_Family.fasta
>
{: .hands_on}
=======
{: .details}

A big step can have several subsections or sub steps:
>>>>>>> Stashed changes

> <question-title></question-title>
>
> 1. Why are we using the tools separately? Can we run it all together?
>
> > <solution-title></solution-title>
> >
> > 1. The tools are run separately to reduce the load on the server and tool. If you have a limited number of taxon names then you can run it all together.
> >
> {: .solution}
>
{: .question}

## Sub-step with **FASTA Merge Files and Filter Unique Sequences**

> <hands-on-title> Task description </hands-on-title>
>
> 1. {% tool [FASTA Merge Files and Filter Unique Sequences](toolshed.g2.bx.psu.edu/repos/galaxyp/fasta_merge_files_and_filter_unique_sequences/fasta_merge_files_and_filter_unique_sequences/1.2.0) %} with the following parameters:
>    - *"Run in batch mode?"*: `Merge individual FASTAs (output collection if the input is collection)`
>        - In *"Input FASTA File(s)"*:
>            - {% icon param-repeat %} *"Insert Input FASTA File(s)"*
>                - {% icon param-file %} *"FASTA File"*: `output` (Input dataset)
>
>    ***TODO***: *Check parameter descriptions*
>
>    ***TODO***: *Consider adding a comment or tip box*
>
>    > <comment-title> short description </comment-title>
>    >
>    > A comment about the tool or something else. This box can also be in the main text
>    {: .comment}
>
{: .hands_on}

<<<<<<< Updated upstream
> <question-title></question-title>
>
> 1. Can we select multiple files together?
=======
***TODO***: *Consider adding a question to test the learners understanding of the previous exercise*

> <question-title></question-title>
>
> 1. Question1?
>>>>>>> Stashed changes
> 2. Question2?
>
> > <solution-title></solution-title>
> >
<<<<<<< Updated upstream
> > 1. Yes, that certainly can be done. We used one input file at a time to maintain the order of sequences in the database.
=======
> > 1. Answer for question1
>>>>>>> Stashed changes
> > 2. Answer for question2
> >
> {: .solution}
>
{: .question}
<<<<<<< Updated upstream

## Sub-step with **Protein Database Downloader**

> <hands-on-title> Task description </hands-on-title>
>
> 1. {% tool [Protein Database Downloader](toolshed.g2.bx.psu.edu/repos/galaxyp/dbbuilder/dbbuilder/0.3.4) %} with the following parameters:
>    - *"Dowload from?"*: `UniProtKB`
>        - In *"Taxonomy"*: `Homo sapiens (Human)`
>	 - In  *"Proteome Set"*: `Reference Proteome Set`
>	 - In  *"Include isoform data"*: `Yes`
> 2. Rename as "Protein Database Human UniProt with Isoforms"
>
{: .hands_on}

## Sub-step with **Protein Database Downloader**

> <hands-on-title> Task description </hands-on-title>
>
> 1. {% tool [Protein Database Downloader](toolshed.g2.bx.psu.edu/repos/galaxyp/dbbuilder/dbbuilder/0.3.4) %} with the following parameters:
>    - *"Dowload from?"*: `cRAP (contaminants)`
> 2. Rename as "Protein Database Contaminants (cRAP)"
>
{: .hands_on}

## Sub-step with **FASTA Merge Files and Filter Unique Sequences**

> <hands-on-title> Task description </hands-on-title>
>
> 1. {% tool [FASTA Merge Files and Filter Unique Sequences](toolshed.g2.bx.psu.edu/repos/galaxyp/fasta_merge_files_and_filter_unique_sequences/fasta_merge_files_and_filter_unique_sequences/1.2.0) %} with the following parameters:
>    - *"Run in batch mode?"*: `Merge individual FASTAs (output collection if input is collection)`
>        - In *"Input FASTA File(s)"*:
>            - {% icon param-repeat %} *"Insert Input FASTA File(s)"*
>                - {% icon param-file %} *"FASTA File"*: `Protein Database Human UniProt with Isoforms` (output of **Protein Database Downloader** {% icon tool %})
>                - {% icon param-file %} *"FASTA File"*: `Protein Database Contaminants (cRAP)` (output of **Protein Database Downloader** {% icon tool %})
>                  
{: .hands_on}

# Reducing Database size
=======
>>>>>>> Stashed changes

## Sub-step with **MetaNovo**

> <hands-on-title> Metanovo tool generates a compact database from your comprehensive database. </hands-on-title>
>
> 1. {% tool [MetaNovo](toolshed.g2.bx.psu.edu/repos/galaxyp/metanovo/metanovo/1.9.4+galaxy4) %} with the following parameters:
>    - *"MGF Input Type"*: `Collection`
>        - {% icon param-collection %} *"MGF Collection"*: `output` (Input dataset collection)
>    - {% icon param-file %} *"FASTA File"*: `output` (output of **FASTA Merge Files and Filter Unique Sequences** {% icon tool %})
>    - In *"Spectrum Matching Parameters"*:
>        - *"Fragment ion mass tolerance"*: `0.01`
>        - *"Enzyme"*: ``
>        - *"Fixed modifications as comma separated list"*: ``
>        - *"Variable modifications as comma separated list"*: ``
>        - *"Maximal charge to search for"*: `5`
>    - In *"Import Filters"*:
>        - *"The maximal peptide length to consider when importing identification files"*: `50`
>
>    ***TODO***: *Check parameter descriptions*
>
>    ***TODO***: *Consider adding a comment or tip box*
>
>    > <comment-title> short description </comment-title>
>    >
>    > A comment about the tool or something else. This box can also be in the main text
>    {: .comment}
>
{: .hands_on}


> <question-title></question-title>
>
<<<<<<< Updated upstream
> 1. Why are we reducing the size of the database?
> 2. Why is this running TMT10 plex modification when the data is 11-plex?
=======
> 1. Question1?
> 2. Question2?
>>>>>>> Stashed changes
>
> > <solution-title></solution-title>
> >
> > 1. Reducing the size of the database improves search speed, FDR and sensitivity.
> > 2. There is no option for 11-plex modifications in Metanovo, hence we use the TMT-10plex.
> >
> {: .solution}
>
{: .question}

<<<<<<< Updated upstream
## Sub-step with **Protein Database Downloader**

> <hands-on-title> Task description </hands-on-title>
> 1. {% tool [Protein Database Downloader](toolshed.g2.bx.psu.edu/repos/galaxyp/dbbuilder/dbbuilder/0.3.4) %} with the following parameters:
>    - *"Dowload from?"*: `UniProtKB(reviewed only)`
>        - In *"Taxonomy"*: `Homo sapiens (Human)`
>	 - In  *"Proteome Set"*: `Reference Proteome Set`
>	 - In  *"Include isoform data"*: `Yes`
> 2. Rename as "Protein Database Human SwissProt".
{: .hands_on}

=======
>>>>>>> Stashed changes
## Sub-step with **FASTA Merge Files and Filter Unique Sequences**

> <hands-on-title> Task description </hands-on-title>
>
> 1. {% tool [FASTA Merge Files and Filter Unique Sequences](toolshed.g2.bx.psu.edu/repos/galaxyp/fasta_merge_files_and_filter_unique_sequences/fasta_merge_files_and_filter_unique_sequences/1.2.0) %} with the following parameters:
>    - *"Run in batch mode?"*: `Merge individual FASTAs (output collection if input is collection)`
>        - In *"Input FASTA File(s)"*:
>            - {% icon param-repeat %} *"Insert Input FASTA File(s)"*
>                - {% icon param-file %} *"FASTA File"*: `output` (Input dataset)
>
>    ***TODO***: *Check parameter descriptions*
>
>    ***TODO***: *Consider adding a comment or tip box*
>
>    > <comment-title> short description </comment-title>
>    >
>    > A comment about the tool or something else. This box can also be in the main text
>    {: .comment}
>
{: .hands_on}


# Conclusion

The first step for the Clinical Metaproteomics study is database generation. As we didnâ€™t have a reference database or information from 16srRNA-seq data, we generated a fasta database doing a literature survey, however, if 16D rRNA data is present, the taxon identified can be used for a customized database generation. As the size of the comprehensive database is generally  too large, we used the Metanovo tool to reduce the size of the database. This reduced database will be then used for clinical metaproteomics discovery workflow.
