<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Galaxy Training!</title>
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <link rel="stylesheet" href="/training-material/assets/css/font-awesome.css">
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />
    </head>
    <body>
        




<header>
    <nav class="navbar navbar-expand-lg navbar-dark">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/topics/proteomics" title="Go back to list of tutorials">
                            <i class="fa fa-folder-o" aria-hidden="true"></i> Proteomics
                        </a>
                    </li>

                    
                        
                        
                        
                    

                    
                        <li class="nav-item dropdown">
                            <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
                                <i class="fa fa-cog" aria-hidden="true"></i> Galaxy Instances
                            </a>
                            <div class="dropdown-menu">
                                
                                    <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/master/topics/proteomics/docker" title="Docker image for this tutorial">
                                        <i class="fa fa-ship" aria-hidden="true"></i> Docker image
                                    </a>
                                
                                
                                
                            </div>
                        </li>
                    

                    

                    
                    <li class="nav-item">
                        <a class="nav-link" href="https://doi.org/10.5281/zenodo.839701" title="Links to data">
                            <i class="fa fa-files-o" aria-hidden="true"></i> Input Dataset
                        </a>
                    </li>
                    

                    
                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/topics/proteomics#references" title="References">
                            <i class="fa fa-book" aria-hidden="true"></i> Literature
                        </a>
                    </li>
                    

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="fa fa-life-ring" aria-hidden="true"></i> Help
    </a>
    <div class="dropdown-menu">
        <a class="dropdown-item" href="/training-material/faq" title="Check our FAQ">
            FAQ
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Chat on Gitter">
            Gitter
        </a>
        <a class="dropdown-item" href="https://biostar.usegalaxy.org/" title="Ask on Biostars">
            Biostars
        </a>
    </div>
</li>


                    <li class="nav-item">
                        <a class="nav-link" href="https://github.com/galaxyproject/training-material/edit/master/topics/proteomics/tutorials/metaproteomics/tutorial.md">
                            <i class="fa fa-github" aria-hidden="true"></i> Edit
                        </a>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

<div class="container main-content">
    <section class="tutorial">
        <h1>Metaproteomics tutorial</h1>

        <blockquote class="overview">
            <h3>Overview</h3>

            <strong><i class="fa fa-question-circle" aria-hidden="true"></i> Questions</strong>
            <ul>
            
            <li>How can I match metaproteomic mass spectrometry data to peptide sequences derived from shotgun metagenomic data?</li>
            
            <li>How can I perform taxonomy analysis and visualize metaproteomics data?</li>
            
            <li>How can I perform functional analysis on this metaproteomics data?</li>
            
            </ul>

            <strong><i class="fa fa-bullseye" aria-hidden="true"></i> Objectives</strong>
            <ul>
            
            <li>A taxonomy and functional analysis of metaproteomic mass spectrometry data.</li>
            
            </ul>

            
            <strong><i class="fa fa-check-circle" aria-hidden="true"></i> Requirements</strong>
            <ul>
            
                <li>
                    
                    <a href="/training-material/topics//introduction/">Galaxy introduction</a>
                    
                </li>
            
            
            </ul>
            

            <p><strong><i class="fa fa-hourglass-end" aria-hidden="true"></i> Time estimation:</strong> 2h</p>
        </blockquote>

        <h1 class="no_toc" id="introduction">Introduction</h1>

<p>In this metaproteomics tutorial we will identify expressed proteins from a complex bacterial community sample.
For this MS/MS data will be matched to peptide sequences provided through a FASTA file.</p>

<p>Metaproteomics is the large-scale characterization of the entire protein complement of environmental microbiota
at a given point in time. It has the potential to unravel the mechanistic details of microbial interactions with
the host / environment by analyzing the functional dynamics of the microbiome.</p>

<p>In this tutorial, we will analyze a sample of sea water that was collected in August of 2013 from the Bering
Strait chlorophyll maximum layer (7m depth, 65° 43.44″ N, 168° 57.42″ W). The data were originally published in <a href="https://www.ncbi.nlm.nih.gov/pubmed/27396978">May et al., 2016</a>.</p>

<blockquote class="agenda">
  <h3 id="agenda">Agenda</h3>

  <p>In this tutorial, we will deal with:</p>

<ol id="markdown-toc">
  <li><a href="#pretreatments" id="markdown-toc-pretreatments">Pretreatments</a>    <ol>
      <li><a href="#data-upload" id="markdown-toc-data-upload">Data upload</a></li>
    </ol>
  </li>
  <li><a href="#analysis" id="markdown-toc-analysis">Analysis</a>    <ol>
      <li><a href="#match-peptide-sequences" id="markdown-toc-match-peptide-sequences">Match peptide sequences</a></li>
      <li><a href="#taxonomy-analysis" id="markdown-toc-taxonomy-analysis">Taxonomy analysis</a></li>
      <li><a href="#genus-taxonomy-level-summary" id="markdown-toc-genus-taxonomy-level-summary">Genus taxonomy level summary</a></li>
      <li><a href="#functional-analysis" id="markdown-toc-functional-analysis">Functional Analysis</a></li>
    </ol>
  </li>
</ol>

</blockquote>

<h1 id="pretreatments">Pretreatments</h1>

<h2 id="data-upload">Data upload</h2>

<p>There are a many ways how you can upload your data. Three among these are:</p>

<ul>
  <li>Upload the files from your computer</li>
  <li>Using a direct link</li>
  <li>Import from the data library if your instance provides the files</li>
</ul>

<p>In this tutorial, we will get the data from Zenodo: <a href="https://doi.org/10.5281/zenodo.839701"><img src="https://zenodo.org/badge/DOI/10.5281/zenodo.839701.svg" alt="DOI" /></a>.</p>

<blockquote class="hands_on">
  <h3 id="-hands-on-data-upload-and-organization"><i class="fa fa-pencil" aria-hidden="true"></i> Hands-on: Data upload and organization</h3>

  <ol>
    <li>Create a new history and name it something meaningful (e.g. <em>Metaproteomics tutorial</em>)</li>
    <li>
      <p>Import the three MGF MS/MS files and the FASTA sequence file from Zenodo.</p>

      <blockquote class="tip">
        <h3 id="-tip-importing-data-via-links"><i class="fa fa-lightbulb-o" aria-hidden="true"></i> Tip: Importing data via links</h3>

        <ul>
          <li>Copy the link location</li>
          <li>Open the Galaxy Upload Manager</li>
          <li>Select <strong>Paste/Fetch Data</strong></li>
          <li>Paste the link into the text field. You can add multiple links, each on a separate line.</li>
          <li>Press <strong>Start</strong></li>
        </ul>
      </blockquote>

      <p>As default, Galaxy takes the link as name.</p>

      <blockquote class="comment">
        <h3 id="-comments"><i class="fa fa-commenting-o" aria-hidden="true"></i> Comments</h3>
        <ul>
          <li>Rename the datasets to a more descriptive name</li>
        </ul>
      </blockquote>
    </li>
    <li>Build a <strong>Dataset list</strong> for the three MGF files
      <ul>
        <li>Click the <strong>Operations on multiple datasets</strong> check box at the top of the history panel
 <img src="../../images/operations_icon.png" alt="Operations on multiple datasets button" /></li>
        <li>Check the three boxes next to the MGF files</li>
        <li>Click <strong>For all selected…</strong> and choose <strong>Build dataset list</strong></li>
        <li>Ensure the three control samples are the only ones selected, and enter a name for the new collection (e.g. <em>MGF files</em>)</li>
        <li>Click <strong>Create list</strong> and exit by clicking again the dataset operations icon</li>
      </ul>
    </li>
  </ol>

</blockquote>

<h1 id="analysis">Analysis</h1>

<h2 id="match-peptide-sequences">Match peptide sequences</h2>

<p>The search database labelled <code class="highlighter-rouge">FASTA_Bering_Strait_Trimmed_metapeptides_cRAP.FASTA</code> is the input database that
will be used to match MS/MS to peptide sequences via a sequence database search. It is a small excerpt of the original database, which was constructed based on a metagenomic screening of the sea water samples (see <a href="https://www.ncbi.nlm.nih.gov/pubmed/27396978">May et al. (2016)</a>). The full original database can be accessed from <a href="https://noble.gs.washington.edu/proj/metapeptide/data/metapeptides_BSt.fasta">here</a>. A contaminant database was added.</p>

<p>For this, the sequence database-searching program called <a href="https://compomics.github.io/projects/searchgui.html">SearchGUI</a> will be used.
The created dataset collection of the three <em>MGF files</em> in the history is used as the MS/MS input.</p>

<h4 id="searchgui">SearchGUI</h4>

<blockquote class="hands_on">
  <h3 id="-hands-on-searchgui"><i class="fa fa-pencil" aria-hidden="true"></i> Hands-on: SearchGUI</h3>

  <ol>
    <li><strong>SearchGUI</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Run <strong>SearchGUI</strong> with:
      <ul>
        <li><strong>Protein Database</strong>: <code class="highlighter-rouge">FASTA_Bering_Strait_Trimmed_metapeptides_cRAP.FASTA</code>(or however you named the <code class="highlighter-rouge">FASTA</code> file)</li>
        <li><strong>Input Peak lists (mgf)</strong>: <code class="highlighter-rouge">MGF files</code> dataset collection.</li>
      </ul>

      <blockquote class="tip">
        <h3 id="-tip-select-dataset-collections-as-input"><i class="fa fa-lightbulb-o" aria-hidden="true"></i> Tip: Select dataset collections as input</h3>

        <ul>
          <li>
            <p>Click the <strong>Dataset collection</strong> icon on the left of the input field:</p>

            <p><img src="../../images/dataset_button.png" alt="Dataset collection button" /></p>
          </li>
          <li>
            <p>Select the appropriate dataset collection from the list</p>
          </li>
        </ul>
      </blockquote>

      <p>Section <strong>Search Engine Options</strong>:</p>

      <ul>
        <li><strong>B-Search Engines</strong>: <code class="highlighter-rouge">X!Tandem</code></li>
      </ul>

      <blockquote class="comment">
        <h3 id="-comment"><i class="fa fa-commenting-o" aria-hidden="true"></i> Comment</h3>

        <p>The section <strong>Search Engine Options</strong> contains a selection of sequence database searching
programs that are available in SearchGUI. Any combination of these programs can be used for
generating PSMs from MS/MS data. For the purpose of this tutorial, <strong>X!Tandem</strong> we will be used.</p>
      </blockquote>

      <p>Section <strong>Precursor Options</strong>:</p>

      <ul>
        <li><strong>Fragment Tolerance Units</strong>: <code class="highlighter-rouge">Daltons</code></li>
        <li><strong>Fragment Tolerance</strong>: <code class="highlighter-rouge">0.02</code>- this is high resolution MS/MS data</li>
        <li><strong>Maximum Charge</strong>: <code class="highlighter-rouge">6</code></li>
      </ul>

      <p>Section <strong>Protein Modification Options</strong>:</p>

      <ul>
        <li><strong>Fixed Modifications</strong>: <code class="highlighter-rouge">Carbamidomethylation of C</code></li>
        <li><strong>Variable modifications</strong>: <code class="highlighter-rouge">Oxidation of M</code></li>
      </ul>

      <blockquote class="tip">
        <h3 id="-tip-search-for-options"><i class="fa fa-lightbulb-o" aria-hidden="true"></i> Tip: Search for options</h3>

        <ul>
          <li>For selection lists, typing the first few letters in the window will filter the available options.</li>
        </ul>
      </blockquote>

      <p>Section <strong>Advanced Options</strong>:</p>
      <ul>
        <li><strong>X!Tandem Options</strong>: <code class="highlighter-rouge">Advanced</code></li>
        <li><strong>X!Tandem: Quick Acetyl</strong>: <code class="highlighter-rouge">No</code></li>
        <li><strong>X!Tandem: Quick Pyrolidone</strong>: <code class="highlighter-rouge">No</code></li>
        <li><strong>X!Tandem: Protein stP Bias</strong>: <code class="highlighter-rouge">No</code></li>
        <li>
          <p><strong>X!Tandem: Maximum Valid Expectation Value</strong>: <code class="highlighter-rouge">100</code></p>
        </li>
        <li>leave everything else as default</li>
      </ul>
    </li>
    <li>Click <strong>Execute</strong>.</li>
  </ol>

</blockquote>

<p>Once the database search is completed, the SearchGUI tool will output a file (called a
SearchGUI archive file) that will serve as an input for the next section, PeptideShaker.</p>

<blockquote class="comment">
  <h3 id="-comment-1"><i class="fa fa-commenting-o" aria-hidden="true"></i> Comment</h3>
  <p>Note that sequence databases used for metaproteomics are usually much larger than the excerpt used in this tutorial. When using large databases, the peptide identification step can take much more time for computation. In metaproteomics, choosing the optimal database is a crucial step of your workflow, for further reading see <a href="https://www.ncbi.nlm.nih.gov/pubmed/27824341">Timmins-Schiffman et al (2017)</a>.</p>

  <p>To learn more about database construction in general, like integrating contaminant databases or using a decoy strategy for FDR searching, please consult our tutorial on <a href="/training-material/topics/proteomics/tutorials/database-handling/tutorial.html">Database Handling</a>.</p>

</blockquote>

<h4 id="peptideshaker">PeptideShaker</h4>

<p><a href="https://compomics.github.io/projects/peptide-shaker.html">PeptideShaker</a> is a post-processing software tool that
processes data from the SearchGUI software tool. It serves to organize the Peptide-Spectral
Matches (PSMs) generated from SearchGUI processing and is contained in the SearchGUI archive.
It provides an assessment of confidence of the data, inferring proteins identified from the
matched peptide sequences and generates outputs that can be visualized by users to interpret
results. PeptideShaker has been wrapped in Galaxy to work in combination with SearchGUI
outputs.</p>

<blockquote class="comment">
  <h3 id="-comment-2"><i class="fa fa-commenting-o" aria-hidden="true"></i> Comment</h3>
  <p>There are a number of choices for different data files that can be generated using
PeptideShaker. A compressed file can be made containing all information needed to view the
results in the standalone PeptideShaker viewer. A <code class="highlighter-rouge">mzidentML</code> file can be created that contains
all peptide sequence matching information and can be utilized by compatible downstream
software. Other outputs are focused on the inferred proteins identified from the PSMs, as well
as phosphorylation reports, relevant if a phosphoproteomics experiment has been undertaken. 
More detailed information on peptide inference using SearchGUI and PeptideShaker can be found in our tutorial on <a href="/training-material/topics/proteomics/tutorials/protein-id-sg-ps/tutorial.html">Peptide and Protein ID</a>.</p>
</blockquote>

<blockquote class="hands_on">
  <h3 id="-hands-on-peptideshaker"><i class="fa fa-pencil" aria-hidden="true"></i> Hands-on: PeptideShaker</h3>

  <ol>
    <li><strong>PeptideShaker</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Run <strong>PeptideShaker</strong> with:
      <ul>
        <li><strong>Compressed SearchGUI results</strong>: The SearchGUI archive file</li>
        <li><strong>Specify Advanced PeptideShaker Processing Options</strong>: <code class="highlighter-rouge">Default Processing Options</code></li>
        <li><strong>Specify Advanced Filtering Options</strong>: <code class="highlighter-rouge">Advanced Filtering Options</code></li>
        <li><strong>Maximum Precursor Error Type</strong>: <code class="highlighter-rouge">Daltons</code></li>
        <li><strong>Specify Contact Information for mzIdendML</strong>: You can leave the default dummy options for now, but feel free to enter custom contact information.</li>
        <li><strong>Include the protein sequences in mzIdentML</strong>: <code class="highlighter-rouge">No</code></li>
        <li><strong>Output options</strong>: Select the <code class="highlighter-rouge">PSM Report</code> (Peptide-Spectral Match) and the <code class="highlighter-rouge">Certificate of Analysis</code></li>
      </ul>

      <blockquote class="comment">
        <h3 id="-comment-3"><i class="fa fa-commenting-o" aria-hidden="true"></i> Comment</h3>

        <p>The <strong>Certificate of Analysis</strong> provides details on all the parameters
used by both SearchGUI and PeptideShaker in the analysis. This can be downloaded from the
Galaxy instance to your local computer in a text file if desired.</p>
      </blockquote>
    </li>
    <li>Click <strong>Execute</strong> and inspect the resulting files after they turned green with the <strong>View data</strong> icon:
 <img src="../../images/view_data_icon.png" alt="View data button" /></li>
  </ol>

</blockquote>

<p>A number of new items will appear in your history, each corresponding to the outputs selected
in the PeptideShaker parameters. Most relevant for this tutorial is the PSM report:</p>

<figure id="figure-1"><img src="../../images/psm_report.png" alt="Display of the PSM report tabular file" title="The PSM report" /><figcaption><span class="figcaption-prefix">Figure 1:</span> The PSM report</figcaption></figure>

<p>Scrolling at the bottom to the left will show the sequence for the PSM that matched to these
metapeptide entries. Column 3 is the sequence matched for each PSM entry. Every PSM is a
new row in the tabular output.</p>

<p>In the following steps of this tutorial, selected portions of this output will be extracted and used for
analysis of the taxonomic make-up of the sample as well as the biochemical functions
represented by the proteins identified.</p>

<h2 id="taxonomy-analysis">Taxonomy analysis</h2>

<p>In the previous section, the genome sequencing and mass spectrometry data from
processing of biological samples was used to identify peptides present in those samples.
Now those peptides are used as evidence to infer which organisms are represented in the sample,
and what biological functions those peptides and associated proteins suggest are occurring.</p>

<p>The UniProt organization collects and annotates all known proteins for organisms. A UniProt
entry includes the protein amino acid sequence, the NCBI taxonomy, and any annotations
about structure and function of the protein. The UniPept web resource developed
by Ghent University will be used to match the sample peptides to proteins. UniPept indexes all Uniprot
proteins and provides a fast matching algorithm for peptides.</p>

<blockquote class="tip">
  <h3 id="-tip-unipept"><i class="fa fa-lightbulb-o" aria-hidden="true"></i> Tip: Unipept</h3>

  <p>Users can access UniPept via a <a href="https://unipept.ugent.be">web page</a> and paste peptide
sequences into the search form to retrieve protein information. But we`ll use a Galaxy
<em>Unipept</em> tool to automate the process. The <em>Unipept</em> tool sends the peptide list to the
UniPept REST API service, then transforms the results into datasets that can be further analyzed
or operated on within Galaxy.</p>
</blockquote>

<h4 id="recieving-the-list-of-peptides-query-tabular">Recieving the list of peptides: Query Tabular</h4>

<p>In order to use <em>Unipept</em>, a list containing the peptide sequences has to be generated.
The tool <strong>Query Tabular</strong> can load tabular data (the PSM report in this case) into a SQLite data base.
As a tabular file is being read, line filters may be applied and an SQL query can be performed.</p>

<blockquote class="hands_on">
  <h3 id="-hands-on-query-tabular"><i class="fa fa-pencil" aria-hidden="true"></i> Hands-on: Query Tabular</h3>

  <ol>
    <li>
      <p><strong>Query Tabular</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Run <strong>Query Tabular</strong> with:</p>

      <ul>
        <li><strong>Database Table</strong>: Click on <code class="highlighter-rouge">+ Insert Database Table</code>:</li>
        <li><strong>Tabular Dataset for Table</strong>: The PSM report</li>
      </ul>

      <p>Section <strong>Filter Dataset Input</strong>:</p>

      <ul>
        <li><strong>Filter Tabular Input Lines</strong>: Click on <code class="highlighter-rouge">+ Insert Filter Tabular Input Lines</code>:</li>
        <li><strong>Filter By</strong>: Select <code class="highlighter-rouge">by regex expression matching</code>
          <ul>
            <li><strong>regex pattern</strong>: <code class="highlighter-rouge">^\d</code></li>
            <li><strong>action for regex match</strong>: <code class="highlighter-rouge">include line on pattern match</code></li>
          </ul>
        </li>
      </ul>

      <p>Section <strong>Table Options</strong>:</p>

      <ul>
        <li><strong>Specify Name for Table</strong>: <code class="highlighter-rouge">psm</code></li>
        <li>
          <p><strong>Specify Column Names (comma-separated list)</strong>: <code class="highlighter-rouge">id,,sequence,,,,,,,,,,,,,,,,,,,,confidence,validation</code></p>

          <blockquote class="comment">
            <h3 id="-comment-4"><i class="fa fa-commenting-o" aria-hidden="true"></i> Comment</h3>

            <p>By default, table columns will be named: c1,c2,c3,…,cn (column names for a table must be unique).
You can override the default names by entering a comma separated list of names, e.g. <code class="highlighter-rouge">,name1,,,name2</code>
would rename the second and fifth columns.</p>

            <p>Check your input file to find the settings which best fits your needs.</p>
          </blockquote>
        </li>
        <li>
          <p><strong>Only load the columns you have named into database</strong>: <code class="highlighter-rouge">Yes</code></p>
        </li>
        <li>
          <p><strong>Save the sqlite database in your history</strong>: <code class="highlighter-rouge">Yes</code></p>

          <blockquote class="tip">
            <h3 id="-tip"><i class="fa fa-lightbulb-o" aria-hidden="true"></i> Tip</h3>

            <ul>
              <li><strong>Query Tabular</strong> can also use an existing SQLite database. Activating <code class="highlighter-rouge">Save the sqlite database in your history</code>
will store the created database in the history, allowing to reuse it directly.</li>
            </ul>

          </blockquote>
        </li>
        <li>
          <p><strong>SQL Query to generate tabular output</strong>:</p>

          <div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>SELECT distinct sequence

FROM psm

WHERE confidence &gt;= 95

ORDER BY sequence
</code></pre></div>          </div>
        </li>
      </ul>

      <blockquote class="question">
        <h3 id="-questions"><i class="fa fa-question-circle" aria-hidden="true"></i> Questions</h3>

        <p>The SQL query might look confusing at first, but having a closer look should clarify a lot.</p>

        <ol>
          <li>What does <code class="highlighter-rouge">FROM psm</code> mean?</li>
          <li>
            <p>What need to be changed if we only want peptides with a confidence higher then 98%?</p>

            <blockquote class="solution">
              <h3 id="-solution"><i class="fa fa-eye" aria-hidden="true"></i> Solution</h3>
              <ol>
                <li>We want to read from table “psm”. We defined the name before in the “Specify Name for Table” option.</li>
                <li>We need to change the value in line 3: “WHERE validation IS NOT ‘Confident’ AND confidence &gt;= 98”</li>
              </ol>
            </blockquote>
          </li>
        </ol>
      </blockquote>

      <ul>
        <li><strong>include query result column headers</strong>: <code class="highlighter-rouge">No</code></li>
      </ul>
    </li>
    <li>
      <p>Click <strong>Execute</strong> and inspect the query results file after it turned green. If everything went well, it should look similiar:</p>

      <figure id="figure-2"><img src="../../images/query_tabular_1.png" alt="Query Tabular output showing the peptides" title="Query Tabular output" /><figcaption><span class="figcaption-prefix">Figure 2:</span> Query Tabular output</figcaption></figure>
    </li>
  </ol>

</blockquote>

<p>While we can proceed with this list of peptides, let’s practice using the created SQLite database for further queries.
We might not only be interested in all the distinct peptides, but also on how many PSMs a single peptide had.
Therefore we can search the database for the peptides and count the occurrence without configuring the tables and columns again:</p>

<blockquote class="hands_on">
  <h3 id="-hands-on-sqlite-to-tabular"><i class="fa fa-pencil" aria-hidden="true"></i> Hands-on: SQLite to tabular</h3>

  <ol>
    <li>
      <p><strong>SQLite to tabular</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Run <strong>SQLite to tabular</strong> with:</p>

      <ul>
        <li>
          <p><strong>SQL Query</strong>:</p>

          <div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>SELECT sequence as "peptide", count(id) as "PSMs"

FROM psm

WHERE confidence &gt;= 95

GROUP BY sequence

ORDER BY sequence
</code></pre></div>          </div>
        </li>
      </ul>
    </li>
    <li>
      <p>Click <strong>Execute</strong>. The resulting file should have two columns, one with the distinct peptides, the other with the count number of PSMs.</p>
    </li>
  </ol>

</blockquote>

<h4 id="retrieve-taxonomy-for-peptides-unipept">Retrieve taxonomy for peptides: Unipept</h4>

<p>The generated list of peptides can now be used to search via <em>Unipept</em>.
We do a taxonomy analysis using the UniPept pept2lca function to return the taxonomic lowest common ancestor for each peptide:</p>

<blockquote class="hands_on">
  <h3 id="-hands-on-unipept"><i class="fa fa-pencil" aria-hidden="true"></i> Hands-on: Unipept</h3>

  <ol>
    <li>
      <p><strong>Unipept</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Run <strong>Unipept</strong> with:</p>

      <ul>
        <li><strong>Unipept application</strong>: <code class="highlighter-rouge">pept2lca: lowest common ancestor</code></li>
        <li><strong>Peptides input format</strong>: <code class="highlighter-rouge">tabular</code></li>
        <li><strong>Tabular Input Containing Peptide column</strong>: The query results file.</li>
        <li><strong>Select column with peptides</strong>: <code class="highlighter-rouge">Column 1</code></li>
        <li><strong>Choose outputs</strong>: Select <code class="highlighter-rouge">tabular</code> and <code class="highlighter-rouge">JSON taxonomy tree</code></li>
      </ul>
    </li>
    <li>
      <p>Click <strong>Execute</strong>. The history should grow by two files. View each to see the difference.</p>

      <blockquote class="comment">
        <h3 id="-comment-5"><i class="fa fa-commenting-o" aria-hidden="true"></i> Comment</h3>

        <p>The JSON (JavaScript Object Notation) file contains the same information as the tabular file but is not comfortably human readable.
Instead, we can use it to use JavaScript libraries to visualize this data.</p>
      </blockquote>
    </li>
    <li>
      <p>Visualize the data:</p>

      <ul>
        <li>
          <p>Click on the JSON output file from the <em>Unipept</em> tool to expand it. Click on the <strong>Visualize</strong> button and select <strong>Unipept Tree viewer</strong>:</p>

          <p><img src="../../images/visualize_button.png" alt="Visualize button" /></p>
        </li>
        <li>
          <p>A new window should appear with a visualization of the taxonomy tree of your data. Use the mouse wheel to scroll in and out and click on nodes to expand or collapse them:</p>

          <figure id="figure-3"><img src="../../images/unipept_tree_viewer.png" alt="Unipept Tree viewer visual output" title="Interactive visualization from the Unipept Tree viever plugin" /><figcaption><span class="figcaption-prefix">Figure 3:</span> Interactive visualization from the Unipept Tree viever plugin</figcaption></figure>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<h2 id="genus-taxonomy-level-summary">Genus taxonomy level summary</h2>

<p>The tabular <em>Unipept</em> output lists the taxonomy assignments for each peptide. To create a meaningful summary, the <strong>Query Tabular</strong> tool is
once again used, aggregating the number of peptides and PSMs for each genus level taxonomy assignment:</p>

<blockquote class="hands_on">
  <h3 id="-hands-on-query-tabular-1"><i class="fa fa-pencil" aria-hidden="true"></i> Hands-on: Query Tabular</h3>

  <ol>
    <li>
      <p><strong>Query Tabular</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Run <strong>Query Tabular</strong> with:</p>

      <ul>
        <li><strong>Database Table</strong>: Click on <code class="highlighter-rouge">+ Insert Database Table</code></li>
        <li><strong>Tabular Dataset for Table</strong>: The PSM report</li>
      </ul>

      <p>Section <strong>Filter Dataset Input</strong>:</p>

      <ul>
        <li><strong>Filter Tabular Input Lines</strong>: Click on <code class="highlighter-rouge">+ Insert Filter Tabular Input Lines</code>:</li>
        <li><strong>Filter By</strong>: Select <code class="highlighter-rouge">by regex expression matching</code>
          <ul>
            <li><strong>regex pattern</strong>: <code class="highlighter-rouge">^\d</code></li>
            <li><strong>action for regex match</strong>: <code class="highlighter-rouge">include line on pattern match</code></li>
          </ul>
        </li>
      </ul>

      <p>Section <strong>Table Options</strong>:</p>

      <ul>
        <li><strong>Specify Name for Table</strong>: <code class="highlighter-rouge">psm</code></li>
        <li>
          <p><strong>Specify Column Names (comma-separated list)</strong>: <code class="highlighter-rouge">,,sequence,,,,,,,,,,,,,,,,,,,,confidence,validation</code></p>
        </li>
        <li><strong>Only load the columns you have named into database</strong>: <code class="highlighter-rouge">Yes</code></li>
      </ul>
    </li>
    <li>
      <p>Repeat this step to have a second <strong>Database Table</strong>:</p>

      <ul>
        <li><strong>Database Table</strong>: Click on <code class="highlighter-rouge">+ Insert Database Table</code></li>
        <li><strong>Tabular Dataset for Table</strong>: The <strong>Unipept</strong> <code class="highlighter-rouge">tabular</code>/<code class="highlighter-rouge">tsv</code> output</li>
      </ul>

      <p>Section <strong>Filter Dataset Input</strong>:</p>

      <ul>
        <li><strong>Filter Tabular Input Lines</strong>: Click on <code class="highlighter-rouge">+ Insert Filter Tabular Input Lines</code>:</li>
        <li><strong>Filter By</strong>: Select <code class="highlighter-rouge">by regex expression matching</code>
          <ul>
            <li><strong>regex pattern</strong>: <code class="highlighter-rouge">#peptide</code></li>
            <li><strong>action for regex match</strong>: <code class="highlighter-rouge">exclude line on pattern match</code></li>
          </ul>
        </li>
      </ul>

      <p>Section <strong>Table Options</strong>:</p>

      <ul>
        <li><strong>Specify Name for Table</strong>: <code class="highlighter-rouge">lca</code></li>
        <li>
          <p><strong>Specify Column Names (comma-separated list)</strong>: <code class="highlighter-rouge">peptide,,,,,,,,,,,,,,,,,,,,,genus</code></p>
        </li>
        <li>
          <p><strong>Only load the columns you have named into database</strong>: <code class="highlighter-rouge">Yes</code></p>
        </li>
        <li>
          <p><strong>Save the sqlite database in your history</strong>: <code class="highlighter-rouge">Yes</code></p>
        </li>
        <li>
          <p><strong>SQL Query to generate tabular output</strong>:</p>

          <div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>SELECT lca.genus,count(psm.sequence) as "PSMs",count(distinct psm.sequence) as "DISTINCT PEPTIDES"

FROM psm LEFT JOIN lca ON psm.sequence = lca.peptide

WHERE confidence &gt;= 95

GROUP BY lca.genus

ORDER BY PSMs desc, 'DISTINCT PEPTIDES' desc
</code></pre></div>          </div>
        </li>
      </ul>
    </li>
    <li>
      <p>Click <strong>Execute</strong> and inspect the query results file after it turned green:</p>

      <figure id="figure-4"><img src="../../images/metaproteomics_summary.png" alt="Query Tabular output showing gene, PSMs and distinct peptides" title="Query Tabular output" /><figcaption><span class="figcaption-prefix">Figure 4:</span> Query Tabular output</figcaption></figure>
    </li>
  </ol>

</blockquote>

<h2 id="functional-analysis">Functional Analysis</h2>

<p>Recent advances in microbiome research indicate that functional characterization via metaproteomics analysis has the potential to accurately
measure the microbial response to perturbations. In particular, metaproteomics enables the estimation of the function of the microbial
community based on expressed microbial proteome.</p>

<p>In the following chapter, a functional analysis will be performed using the <strong>UniPept</strong> application <code class="highlighter-rouge">pept2prot</code> in order to match the list of peptides with the correlated Gene Ontology terms.
This allows to get an insight of the <strong>biological process</strong>, the <strong>molecular function</strong> and the <strong>cellular component</strong> related to the sample data.</p>

<blockquote class="comment">
  <h3 id="-gene-ontology-consortium"><i class="fa fa-commenting-o" aria-hidden="true"></i> Gene Ontology Consortium</h3>

  <p>The <a href="http://www.geneontology.org/">Gene Ontology Consortium</a> provides with its Ontology a framework for the model of biology.
The GO defines concepts/classes used to describe gene function, and relationships between these concepts. It classifies functions along three aspects:</p>

  <ul>
    <li>
      <p><strong>molecular function</strong></p>

      <ul>
        <li>molecular activities of gene products</li>
      </ul>
    </li>
    <li>
      <p><strong>cellular component</strong></p>

      <ul>
        <li>where gene products are active</li>
      </ul>
    </li>
    <li>
      <p><strong>biological process</strong></p>

      <ul>
        <li>pathways and larger processes made up of the activities of multiple gene products.</li>
      </ul>
    </li>
  </ul>

  <p><a href="http://geneontology.org/page/ontology-documentation">more information</a></p>

</blockquote>

<h4 id="data-upload-1">Data upload</h4>

<p>For this tutorial, a tabular file containing the relevant GO terms has been created. It contains the GO aspect, the ID and the name.
It is available at Zenodo: <a href="https://doi.org/10.5281/zenodo.839701"><img src="https://zenodo.org/badge/DOI/10.5281/zenodo.839701.svg" alt="DOI" /></a>.</p>

<blockquote class="hands_on">
  <h3 id="-hands-on-data-upload"><i class="fa fa-pencil" aria-hidden="true"></i> Hands-on: Data upload</h3>

  <ol>
    <li>
      <p>Import the file <code class="highlighter-rouge">Gene_Ontology_Terms.tabular</code> from Zenodo.</p>

      <blockquote class="tip">
        <h3 id="-tip-setting-file-metadata-on-upload"><i class="fa fa-lightbulb-o" aria-hidden="true"></i> Tip: Setting file metadata on upload</h3>

        <p>In the upload window of Galaxy you can set the filetype and related genome of the file you’re uploading in the corresponding columns beforehand.
This might be handy if the automatic detection of the filetype didn’t work out perfectly or if you want to avoid setting the genome later on, especially for multiple files.</p>

      </blockquote>

      <p>As default, Galaxy takes the link as name.</p>

      <blockquote class="comment">
        <h3 id="-comments-1"><i class="fa fa-commenting-o" aria-hidden="true"></i> Comments</h3>
        <ul>
          <li>Rename the datasets to a more descriptive name, e.g. <code class="highlighter-rouge">Gene Ontology Terms</code></li>
        </ul>
      </blockquote>
    </li>
  </ol>

</blockquote>

<blockquote class="tip">
  <h3 id="-tip-creating-your-own-gene-ontology-list"><i class="fa fa-lightbulb-o" aria-hidden="true"></i> Tip: Creating your own Gene Ontology list</h3>

  <p>The latest Gene Ontology can be downloaded <a href="http://geneontology.org/page/download-ontology">here</a> as a text file in the <code class="highlighter-rouge">OBO</code> format.
<code class="highlighter-rouge">OBO</code> files are human-readable (in addition to machine-readable) and can be opened in any text editor. They contain more information than just the name and aspect.</p>

  <p>In order to receive a file like we use in the tutorial for your own analysis, different tools are available to extract information from <code class="highlighter-rouge">OBO</code> files,
one of them being <a href="https://doi.org/10.1093/bioinformatics/btn042">ONTO-PERL</a>.
An example file with all GO terms from 08.07.2017 named <code class="highlighter-rouge">Gene_Ontology_Terms_full_07.08.2017.tabular</code> can be found on the <a href="https://doi.org/10.5281/zenodo.839701">Zenodo repository</a> of this tutorial as well.</p>

</blockquote>

<h4 id="retrieve-go-ids-for-peptides-unipept">Retrieve GO IDs for peptides: Unipept</h4>

<p>The <strong>UniPept</strong> application <code class="highlighter-rouge">pept2prot</code> can be used to return the list of proteins containing each peptide.
The option <code class="highlighter-rouge">retrieve extra information</code> option is set to <code class="highlighter-rouge">yes</code> so that we retrieve Gene Ontology assignments (<code class="highlighter-rouge">go_references</code>)
for each protein.</p>

<blockquote class="hands_on">
  <h3 id="-hands-on-unipept-1"><i class="fa fa-pencil" aria-hidden="true"></i> Hands-on: Unipept</h3>

  <ol>
    <li>
      <p><strong>Unipept</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Run <strong>Unipept</strong> with:</p>

      <ul>
        <li><strong>Unipept application</strong>: <code class="highlighter-rouge">pept2prot: UniProt entries containing a given tryptic peptide</code></li>
        <li><strong>retrieve extra information</strong>: <code class="highlighter-rouge">Yes</code></li>
        <li><strong>Peptides input format</strong>: <code class="highlighter-rouge">tabular</code></li>
        <li><strong>Tabular Input Containing Peptide column</strong>: The first query results file.</li>
        <li><strong>Select column with peptides</strong>: <code class="highlighter-rouge">Column 1</code></li>
        <li><strong>Choose outputs</strong>: Select <code class="highlighter-rouge">tabular</code></li>
      </ul>
    </li>
    <li>
      <p>Click <strong>Execute</strong>.</p>
    </li>
    <li>
      <p>inspect the result:</p>

      <ul>
        <li>The output should be a tabular file containing a column labeled <code class="highlighter-rouge">go_references</code>. This is what we’re looking for.</li>
      </ul>
    </li>
  </ol>

</blockquote>

<h4 id="combine-all-information-to-quantify-the-go-results">Combine all information to quantify the GO results</h4>

<p>As a final step we will use <strong>Query Tabular</strong> in a more sophisticated way to combine all information to quantify the GO analysis. The three used file and the extracted information are:</p>

<ul>
  <li><strong>Gene Ontology Terms</strong>:
    <ul>
      <li><code class="highlighter-rouge">go_id</code> to match with <strong>Normalized UniPept output</strong></li>
      <li>The GO <code class="highlighter-rouge">aspect</code> to group the results in three separate files</li>
      <li>The GO <code class="highlighter-rouge">description</code> to annotate the results</li>
    </ul>
  </li>
  <li><strong>Normalized UniPept output</strong>:
    <ul>
      <li><code class="highlighter-rouge">peptide</code> to match with <strong>PSM Report</strong> and to count distinct peptides per GO term</li>
      <li><code class="highlighter-rouge">go_reference</code> to match with <strong>Gene Ontology Terms</strong></li>
    </ul>
  </li>
  <li><strong>PSM Report</strong>:
    <ul>
      <li><code class="highlighter-rouge">sequence</code> to match with <strong>Normalized UniPept output</strong></li>
      <li><code class="highlighter-rouge">id</code> to count distinct PSM’s per GO term</li>
    </ul>
  </li>
</ul>

<blockquote class="hands_on">
  <h3 id="-hands-on-query-tabular-2"><i class="fa fa-pencil" aria-hidden="true"></i> Hands-on: Query Tabular</h3>

  <ol>
    <li>
      <p><strong>Query Tabular</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Run <strong>Query Tabular</strong> with:</p>

      <ul>
        <li><strong>Database Table</strong>: Click on <code class="highlighter-rouge">+ Insert Database Table</code></li>
        <li><strong>Tabular Dataset for Table</strong>: The <code class="highlighter-rouge">Gene Ontology Terms</code> file</li>
      </ul>

      <p>Section <strong>Filter Dataset Input</strong>:</p>

      <ul>
        <li><strong>Filter Tabular Input Lines</strong>: Click on <code class="highlighter-rouge">+ Insert Filter Tabular Input Lines</code>:</li>
        <li><strong>Filter By</strong>: Select <code class="highlighter-rouge">skip leading lines</code>
          <ul>
            <li><strong>Skip lines</strong>: <code class="highlighter-rouge">1</code></li>
          </ul>
        </li>
      </ul>

      <p>Section <strong>Table Options</strong>:</p>

      <ul>
        <li><strong>Specify Name for Table</strong>: <code class="highlighter-rouge">go</code></li>
        <li><strong>Specify Column Names (comma-separated list)</strong>: <code class="highlighter-rouge">aspect,go_id,description</code></li>
        <li><strong>Table Index</strong>: Click on <code class="highlighter-rouge">+ Insert Table Index</code>:
          <ul>
            <li><strong>This is a unique index</strong>: <code class="highlighter-rouge">No</code></li>
            <li><strong>Index on Columns</strong>: <code class="highlighter-rouge">aspect,go_id</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p>Repeat this step to have a second <strong>Database Table</strong>:</p>

      <ul>
        <li><strong>Database Table</strong>: Click on <code class="highlighter-rouge">+ Insert Database Table</code></li>
        <li><strong>Tabular Dataset for Table</strong>: The <strong>Unipept</strong> <code class="highlighter-rouge">tabluar</code>/<code class="highlighter-rouge">tsv</code> output</li>
      </ul>

      <p>Section <strong>Filter Dataset Input</strong>:</p>

      <ul>
        <li><strong>Filter Tabular Input Lines</strong>: Click on <code class="highlighter-rouge">+ Insert Filter Tabular Input Lines</code>:</li>
        <li><strong>Filter By</strong>: Select <code class="highlighter-rouge">skip leading lines</code>
          <ul>
            <li><strong>Skip lines</strong>: <code class="highlighter-rouge">1</code></li>
          </ul>
        </li>
        <li>Add another Filter: Click on <code class="highlighter-rouge">+ Insert Filter Tabular Input Lines</code>:</li>
        <li><strong>Filter By</strong>: Select <code class="highlighter-rouge">prepend a line number column</code></li>
      </ul>

      <p>Section <strong>Table Options</strong>:</p>

      <ul>
        <li><strong>Specify Name for Table</strong>: <code class="highlighter-rouge">bering_prot</code></li>
        <li><strong>Specify Column Names (comma-separated list)</strong>: <code class="highlighter-rouge">id,peptide,uniprot_id,taxon_id,taxon_name,ec_references,go_references,refseq_ids,refseq_protein_ids,insdc_ids,insdc_protein_ids</code></li>
        <li><strong>Table Index</strong>: Click on <code class="highlighter-rouge">+ Insert Table Index</code>:
          <ul>
            <li><strong>This is a unique index</strong>: <code class="highlighter-rouge">No</code></li>
            <li><strong>Index on Columns</strong>: <code class="highlighter-rouge">id,peptide</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p>Repeat this step to have another <strong>Database Table</strong>:</p>

      <ul>
        <li><strong>Database Table</strong>: Click on <code class="highlighter-rouge">+ Insert Database Table</code></li>
        <li><strong>Tabular Dataset for Table</strong>: The same <strong>Unipept</strong> <code class="highlighter-rouge">tabluar</code>/<code class="highlighter-rouge">tsv</code> output</li>
      </ul>

      <p>Section <strong>Filter Dataset Input</strong>:</p>

      <ul>
        <li><strong>Filter Tabular Input Lines</strong>: Click on <code class="highlighter-rouge">+ Insert Filter Tabular Input Lines</code>:</li>
        <li><strong>Filter By</strong>: Select <code class="highlighter-rouge">skip leading lines</code>
          <ul>
            <li><strong>Skip lines</strong>: leave blank</li>
          </ul>
        </li>
        <li>Add another Filter: Click on <code class="highlighter-rouge">+ Insert Filter Tabular Input Lines</code>:</li>
        <li><strong>Filter By</strong>: Select <code class="highlighter-rouge">prepend a line number column</code></li>
        <li>Add another Filter: Click on <code class="highlighter-rouge">+ Insert Filter Tabular Input Lines</code>:</li>
        <li><strong>Filter By</strong>: Select <code class="highlighter-rouge">select columns</code>
          <ul>
            <li><strong>enter column numbers to keep</strong>: <code class="highlighter-rouge">1,7</code></li>
          </ul>
        </li>
        <li>Add another Filter: Click on <code class="highlighter-rouge">+ Insert Filter Tabular Input Lines</code>:</li>
        <li><strong>Filter By</strong>: Select <code class="highlighter-rouge">normalize list columns, replicates row for each item in list</code>
          <ul>
            <li><strong>enter column numbers to normalize</strong>: <code class="highlighter-rouge">2</code></li>
            <li><strong>List item delimiter in column</strong>: ` ` (a single blank character)</li>
          </ul>
        </li>
      </ul>

      <blockquote class="comment">
        <h3 id="-comments-2"><i class="fa fa-commenting-o" aria-hidden="true"></i> Comments</h3>
        <ul>
          <li>The UniPept result file can contain multiple GO IDs in a single row. In order to create a normalized table of this data, these rows will be split so each record contains only one GO ID.</li>
        </ul>
      </blockquote>

      <p>Section <strong>Table Options</strong>:</p>

      <ul>
        <li><strong>Specify Name for Table</strong>: <code class="highlighter-rouge">bering_prot_go</code></li>
        <li><strong>Specify Column Names (comma-separated list)</strong>: <code class="highlighter-rouge">id,go_reference</code></li>
        <li><strong>Table Index</strong>: Click on <code class="highlighter-rouge">+ Insert Table Index</code>:
          <ul>
            <li><strong>This is a unique index</strong>: <code class="highlighter-rouge">No</code></li>
            <li><strong>Index on Columns</strong>: <code class="highlighter-rouge">go_reference,id</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p>Repeat this step to have another <strong>Database Table</strong>:</p>

      <ul>
        <li><strong>Database Table</strong>: Click on <code class="highlighter-rouge">+ Insert Database Table</code></li>
        <li><strong>Tabular Dataset for Table</strong>: The <code class="highlighter-rouge">PSM Report</code></li>
      </ul>

      <p>Section <strong>Filter Dataset Input</strong>:</p>

      <ul>
        <li><strong>Filter Tabular Input Lines</strong>: Click on <code class="highlighter-rouge">+ Insert Filter Tabular Input Lines</code>:</li>
        <li><strong>Filter By</strong>: Select <code class="highlighter-rouge">by regex expression matching</code>
          <ul>
            <li><strong>regex pattern</strong>: <code class="highlighter-rouge">^\d</code></li>
            <li><strong>action for regex match</strong>: <code class="highlighter-rouge">include line on pattern match</code></li>
          </ul>
        </li>
        <li>Add another Filter: Click on <code class="highlighter-rouge">+ Insert Filter Tabular Input Lines</code>:</li>
        <li><strong>Filter By</strong>: Select <code class="highlighter-rouge">select columns</code>
          <ul>
            <li><strong>enter column numbers to keep</strong>: <code class="highlighter-rouge">1,3,23,24</code></li>
          </ul>
        </li>
      </ul>

      <p>Section <strong>Table Options</strong>:</p>

      <ul>
        <li><strong>Specify Name for Table</strong>: <code class="highlighter-rouge">bering_psms</code></li>
        <li><strong>Specify Column Names (comma-separated list)</strong>: <code class="highlighter-rouge">id,sequence,confidence,validation</code></li>
        <li><strong>Only load the columns you have named into database</strong>: <code class="highlighter-rouge">Yes</code></li>
        <li><strong>Table Index</strong>: Click on <code class="highlighter-rouge">+ Insert Table Index</code>:
          <ul>
            <li><strong>This is a unique index</strong>: <code class="highlighter-rouge">No</code></li>
            <li><strong>Index on Columns</strong>: <code class="highlighter-rouge">sequence,id</code></li>
          </ul>
        </li>
        <li>
          <p><strong>Save the sqlite database in your history</strong>: <code class="highlighter-rouge">Yes</code></p>
        </li>
        <li>
          <p><strong>SQL Query to generate tabular output</strong>:</p>

          <div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>SELECT sequence as "peptide", count(id) as "PSMs"

FROM bering_psms

WHERE confidence &gt;= 95

GROUP BY sequence

ORDER BY sequence
</code></pre></div>          </div>
        </li>
      </ul>
    </li>
    <li>
      <p>Click <strong>Execute</strong>.</p>
    </li>
  </ol>

</blockquote>

<p>With this we have combined all the data into a single database which we can now query to extract the desired information with <strong>SQLite to tabular</strong>:</p>

<blockquote class="hands_on">
  <h3 id="-hands-on-sqlite-to-tabular-1"><i class="fa fa-pencil" aria-hidden="true"></i> Hands-on: SQLite to tabular</h3>

  <ol>
    <li>
      <p><strong>SQLite to tabular</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Run <strong>SQLite to tabular</strong> with:</p>

      <ul>
        <li><strong>SQLite Database</strong>: The created SQLite database from the former step</li>
        <li>
          <p><strong>SQL Query</strong>:</p>

          <div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>SELECT go.description, 

count(distinct bering_psms.sequence) as "bering_peptides", count(distinct bering_psms.id) as "bering_psms" 

FROM go JOIN bering_prot_go ON go.go_id = bering_prot_go.go_reference JOIN bering_prot on bering_prot_go.id = bering_prot.id JOIN 

bering_psms ON bering_prot.peptide = bering_psms.sequence

WHERE go.aspect = 'molecular_function'

GROUP BY go.description

ORDER BY  bering_peptides desc,bering_psms desc
</code></pre></div>          </div>
        </li>
      </ul>
    </li>
    <li>Click <strong>Execute</strong>.</li>
    <li>Repeat these steps two times by replacing <code class="highlighter-rouge">molecular_function</code> in the fifth row of the SQL query by <code class="highlighter-rouge">biological_process</code> and <code class="highlighter-rouge">cellular_component</code>.</li>
  </ol>

</blockquote>

<p>With these three resulting files the functional analysis of this tutorial is finished. Each record contains the name of a GO term, the amount of peptides related to it and the amount of PSMs for these peptides.</p>

<blockquote class="comment">
  <h3 id="-references"><i class="fa fa-commenting-o" aria-hidden="true"></i> References</h3>

  <ul>
    <li>
      <p><a href="https://www.ncbi.nlm.nih.gov/pubmed/27824341">Dataset</a> and <a href="https://www.ncbi.nlm.nih.gov/pubmed/27396978">SixGill software</a></p>
    </li>
    <li>
      <p><a href="https://www.ncbi.nlm.nih.gov/pubmed/26058579">Galaxy workflows for metaproteomics</a></p>
    </li>
    <li>
      <p><a href="https://z.umn.edu/gcc2017mporal">Metaproteomics community effort</a></p>
    </li>
    <li>
      <p><a href="https://www.ncbi.nlm.nih.gov/pubmed/28552653">Unipept</a></p>
    </li>
  </ul>

</blockquote>



        
        <blockquote class="key_points">
            <h3><i class="fa fa-key" aria-hidden="true"></i> Key points</h3>

            <ul>
                
                <li>Use dataset collections</li>
                
                <li>With SearchGUI and PeptideShaker you can gain access to multiple search engines</li>
                
                <li>Learning the basics of SQL queries can pay off</li>
                
            </ul>
        </blockquote>
        

        
        <h1>Useful literature</h1>
        <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/proteomics#references">here</a>.</p>
        

        

        <h3><i class="fa fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!</h3>

        <hr>

        <blockquote class="overview">
            <h3><i class="fa fa-comments-o" aria-hidden="true"></i> Help us improve this content!</h3>
            Please take a moment to fill in the Galaxy Training Network
            <a href="https://tinyurl.com/GTNfeedback">Feedback Form</a>.
            Your feedback helps us improve this tutorial and will be considered
            in future revisions.
        </blockquote>
    </section>
</div>


<footer>
    <div class="container">
        <p>
            This material is the result of a collaborative work. Thanks to the
            <a href="https://wiki.galaxyproject.org/Teach/GTN">Galaxy Training Network</a>
            and all the <a href="/training-material/hall-of-fame">contributors</a> (Timothy J. Griffin, Pratik Jagtap, James Johnson, Clemens Blank)!
        </p>
        <p>
            Found a typo? Something is wrong in this tutorial? Edit it on
            <a href="https://github.com/galaxyproject/training-material/tree/master/topics/proteomics/tutorials/metaproteomics/tutorial.md">GitHub</a>.
        </p>
    </div>
</footer>

    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>
</html>
